### ISSUE TITLE
No preconditioner specification for spline operator creates exception

### ISSUE THREAD

**Issue by sslattery** [ORNL-CEES/DataTransferKit#122]
In updating CASL to version 2 the point cloud operators tests were found to fail on an exception

```
uy7@u233.ornl.gov:~/localhome/BUILDS/CHECKIN/MPI_RELEASE_DEBUG_SHARED/DataTransferKit/packages/Operators/test$ ./DataTransferKitOperators_PointCloudOperators_test.exe
Teuchos::GlobalMPISession::GlobalMPISession(): started processor with name u233.ornl.gov and rank 0!

***
*** Unit test suite ...
***


Sorting tests by group name then by the order they were added ... (time = 0.000421)

Running unit tests ...

0. MovingLeastSquareReconstructionOperator_mls_radius_test_UnitTest ... [Passed] (0.0853 sec)
1. MovingLeastSquareReconstructionOperator_mls_knn_test_UnitTest ... [Passed] (0.066 sec)
2. SplineInterpolationOperator_spline_radius_test_UnitTest ...

 p=0: *** Caught standard std::exception of type 'Teuchos::m_bad_cast' :

  /localhome/uy7/VERA/Trilinos/packages/teuchos/core/src/Teuchos_dyn_cast.cpp:61:

  Throw number = 2

  Throw test that evaluated to true: true

  dyn_cast<Thyra::EpetraLinearOpBase>(Thyra::LinearOpBase<double>) : Error, the object with the concrete type 'Thyra::DefaultAddedLinearOp<double>' (passed in through the interface type 'Thyra::LinearOpBase<double>')  does not support the interface 'Thyra::EpetraLinearOpBase' and the dynamic cast failed!
 [FAILED]  (0.177 sec) SplineInterpolationOperator_spline_radius_test_UnitTest
 Location: /localhome/uy7/VERA/DataTransferKit/packages/Operators/test/tstPointCloudOperators.cpp:177

3. SplineInterpolationOperator_spline_knn_test_UnitTest ...

 p=0: *** Caught standard std::exception of type 'Teuchos::m_bad_cast' :

  /localhome/uy7/VERA/Trilinos/packages/teuchos/core/src/Teuchos_dyn_cast.cpp:61:

  Throw number = 3

  Throw test that evaluated to true: true

  dyn_cast<Thyra::EpetraLinearOpBase>(Thyra::LinearOpBase<double>) : Error, the object with the concrete type 'Thyra::DefaultAddedLinearOp<double>' (passed in through the interface type 'Thyra::LinearOpBase<double>')  does not support the interface 'Thyra::EpetraLinearOpBase' and the dynamic cast failed!
 [FAILED]  (0.181 sec) SplineInterpolationOperator_spline_knn_test_UnitTest
 Location: /localhome/uy7/VERA/DataTransferKit/packages/Operators/test/tstPointCloudOperators.cpp:195


The following tests FAILED:
    2. SplineInterpolationOperator_spline_radius_test_UnitTest ...
    3. SplineInterpolationOperator_spline_knn_test_UnitTest ...

Total Time: 0.51 sec

Summary: total = 4, run = 4, passed = 2, failed = 2

End Result: TEST FAILED
```

Further investigation showed that Ifpack is the default preconditioner factory in Thyra - therefore casts are made to Epetra objects. This obviously fails on DTK Tpetra objects. This will not get called if the `Preconditioner Type` field in the Stratimikos parameter list is set to none. We don't need a preconditioner for GMRES in the spline operator so I propose automatically setting this in the operator so this exception is never encountered.

**Comment by sslattery** [comment]
Also this affects version 2

**Comment by sslattery** [comment]
Fixed by merge of #123
